### PR TITLE
test: align data providers with test-writing rules

### PR DESCRIPTION
**Pull request by keulinho**
### 1. Why is this change necessary?

The test-writing rules prefer named data-provider cases so failures are easier to understand and individual scenarios document their intent. Several existing unit and integration tests still used anonymous array providers or generated case names that described only raw values.

The migration tests also did not consistently cover `getCreationTimestamp()`, so a timestamp mismatch between a migration class and its filename could slip through without a focused assertion.

### 2. What does this change do, exactly?

This converts affected unit and integration test data providers to named `yield` cases with human-readable descriptions, keeps valid edge cases, and removes only duplicate provider cases that did not add coverage.

It also adds coverage for migration creation timestamps so the timestamp returned by each migration is asserted in its corresponding migration test.

### 3. Describe each step to reproduce the issue or behaviour.

1. Inspect provider-based tests that return anonymous array cases or cases named after raw values.
2. Run one of those tests with a failing provider case and the failing dataset is hard to map back to the scenario.
3. Inspect migration tests and note that `getCreationTimestamp()` values were not asserted consistently.

### 4. Please link to the relevant issues (if any).

None.

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [ ] I have updated developer-facing release notes if this change is **relevant** for external developers:
  - Add a short entry to `RELEASE_INFO-6.<major>.md` under “Upcoming” for informational changes, including the consequences of the change and how it affects external developers.
  - Add an `UPGRADE` section in `UPGRADE-6.<next-major>.md` for breaking changes (what/why/impact/how to adapt).
  - See the [Documenting a Release Process](https://github.com/shopware/shopware/blob/trunk/delivery-process/documenting-a-release.md) for details.
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [ ] I have read the contribution requirements and fulfilled them

### CODE REVIEW

**Review by SpiGAndromeda — APPROVED** [review]

**Review by nfortier-shopware — APPROVED** [review]

File: tests/integration/Administration/Notification/Repository/NotificationRepositoryTest.php [hunk 1]
```diff
@@ -82,14 +82,12 @@ public function testNotificationCreateNotification(string $scope, string $behavi
     }
 
     /**
-     * @return array<int, array<int, string>>
+     * @return iterable<string, array<int, string>>
      */
-    public static function notificationProvider(): array
+    public static function notificationProvider(): iterable
```
Comment (mitelg) [inline tests/integration/Administration/Notification/Repository/NotificationRepositoryTest.php:87]: why not type-hinting `\Generator`? 🙈
